### PR TITLE
avoid %lu since size is not consistent

### PR DESCRIPTION
**Pull request by vjardin**
Let' use PRIu64 formatting so the length will be the same on both 64 bits and 32 bits system.

### CODE REVIEW

**Comment by vjardin** [comment]
fix of https://github.com/tpm2-software/tpm2-pkcs11/issues/805

**Comment by vjardin** [comment]
Please, @williamcroberts , what does the CI keeps failing ? 
For instance:
```
docker: Error response from daemon: Head "https://ghcr.io/v2/tpm2-software/fedora-30/manifests/latest": unauthorized.
```
I do not understand how to work around it.

**Comment by vjardinMisc** [comment]
Please, @williamcroberts , can you check the following for fixing the CI ?
  https://github.com/nektos/act/issues/1248#issuecomment-1178309705

or can you check your PAT (Personal Access Token)? It may have expired.
I believe, it is yours, not mine ; but it is unclear.

**Comment by williamcroberts** [comment]
> Please, @williamcroberts , can you check the following for fixing the CI ? [nektos/act#1248 (comment)](https://github.com/nektos/act/issues/1248#issuecomment-1178309705)
> 
> or can you check your PAT (Personal Access Token)? It may have expired. I believe, it is yours, not mine ; but it is unclear.

I cleaned up our CI images and thought fedora-30 was unused, I restored the image.

File: src/lib/emitter.c [hunk 1]
```diff
@@ -82,7 +83,7 @@ WEAK char *emit_attributes_to_string(attr_list *attrs) {
 
         char strkey[64] = { 0 };
         CK_ATTRIBUTE_TYPE k = a->type;
-        snprintf(strkey, sizeof(strkey), "%lu", k);
+        snprintf(strkey, sizeof(strkey), "%"PRIu64"", k);
```
Comment (williamcroberts) [inline src/lib/emitter.c:86]: on 32 bit systems this causes a warning:
```
a.c: In function 'main':
a.c:6:9: warning: format '%llu' expects argument of type 'long long unsigned int', but argument 2 has type 'long unsigned int' [-Wformat=]
    6 |  printf("%"PRIu64"\n", x);
      |         ^~~            ~
```

**Review by williamcroberts — CHANGES_REQUESTED** [review]
The type is unsigned long, so converting to and from the type should use the `%lu` format string. Their will be no loss of precision because the data type is of that precision and will be written out and read in from the YAML in the precision that is supported.

```C
typedef unsigned long ck_attribute_type_t;
#define ck_attribute_type_t CK_ATTRIBUTE_TYPE
```

Nack, this doesn't fix anything, this actually introduces issues.

**Comment by vjardin** [comment]
> The type is unsigned long, so converting to and from the type should use the `%lu` format string. Their will be no loss of precision because the data type is of that precision and will be written out and read in from the YAML in the precision that is supported.
> 
> ```c
> typedef unsigned long ck_attribute_type_t;
> #define ck_attribute_type_t CK_ATTRIBUTE_TYPE
> ```
> 
> Nack, this doesn't fix anything, this actually introduces issues.

yes, but if you write 32bits and a 64 bits application read it you get some issues.

You write:
  "030000000900000001000000060000004000000041000000420000000d0000000e000000430000004400000045000000"
but you'd like to read:
  "03000000000000000900000000000000010000000000000006000000000000004000000000000000410000000000000042000000000000000d000000000000000e00000000000000430000000000000044000000000000004500000000000000"

File: src/lib/emitter.c (hunk 1 above)
Comment (vjardin) [reply src/lib/emitter.c:86]: Back to the basics, we could enforce a cast:
```
snprintf(strkey, sizeof(strkey), "%"PRIu64"", (u_int64_t)k);
```
but first, let's make sure we agree on the problem.

**Comment by vjardin** [comment]
Per the comments https://github.com/tpm2-software/tpm2-pkcs11/issues/805#issuecomment-1374878320 ; it is another problem to be fixed ! yaml seq vs yaml string ?! for a CKA_ALLOWED_MECHANISMS

**Comment by williamcroberts** [comment]
> Per the comments [#805 (comment)](https://github.com/tpm2-software/tpm2-pkcs11/issues/805#issuecomment-1374878320) ; it is another problem to be fixed ! yaml seq vs yaml string ?! for a CKA_ALLOWED_MECHANISMS

CKA_ALLOWED_MECHANISMS should be a sequence:

```bash
tpm2_ptool objmod --key=CKA_ALLOWED_MECHANISMS --id=12 --path=$HOME/tmp
CKA_ALLOWED_MECHANISMS:
- 4161
- 4162
- 4164
- 4165
- 4166
```
Are you getting a YAML file generated where that's not the case? If so, how can I reproduce it and whats the contents of the whole YAML data? You can use objmod to dump it.

```bash
tpm2_ptool objmod --id=12 --path=$HOME/tmp
0: 3
1: true
2: true
3: '656330'
256: 3
257: ''
258: '65656636653364613638326362633862'
259: true
261: true
<snip>
```

**Pull request closed without merging**